src/solvers/dg/lobatto_legendre.jl

```diff
@@ -0,0 +1,553 @@
+
+struct LobattoLegendreBasis{RealT<:Real, NNODES,
+                            InverseVandermondeLegendre<:AbstractMatrix{RealT},
+                            BoundaryMatrix<:AbstractMatrix{RealT},
+                            DerivativeMatrix<:AbstractMatrix{RealT}} <: AbstractBasisSBP{RealT}
+  nodes          ::SVector{NNODES, RealT}
+  weights        ::SVector{NNODES, RealT}
+  inverse_weights::SVector{NNODES, RealT}
+
+  inverse_vandermonde_legendre::InverseVandermondeLegendre
+  boundary_interpolation      ::BoundaryMatrix # lhat
+
+  derivative_matrix         ::DerivativeMatrix # dsplit
+  derivative_split          ::DerivativeMatrix # dsplit
+  derivative_split_transpose::DerivativeMatrix # dsplit_transposed
+  derivative_neg_adjoint    ::DerivativeMatrix # dhat, neg. adjoint wrt the SBP dot product
```

Comment:
What does `hat` mean? That seems to be very specific to DG methods based on the book of Kopriva. For example, I wouldn't have known what `derivative_dhat` could be without having read this book (or something similar). In contrast, the concept of an `adjoint` is used throughout mathematics and a "negative adjoint" of something seems to be more clear to me.

Reply:
> That seems to be very specific to DG methods based on the book of Kopriva.

Haha, that would have been an argument by myself *for* using `dhat` (since we use it throughout our papers as well). Alas, I don't have a beef with the negative adjoint, but I'd like to ping @gregorgassner such that he sees this as well (and maybe weighs in...).